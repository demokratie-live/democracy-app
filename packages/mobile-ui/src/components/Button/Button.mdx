---
name: Button
route: /button
---

import { Playground, Props } from 'docz';
import { Text } from 'react-native';
import { Button } from './';

# Button

<Props of={Button} />

## Basic usage

<Playground>
  <Button
    onPress={() => {}}
    text="Hello Button"
    textColor="red"
    backgroundColor="blue"
  />
</Playground>
