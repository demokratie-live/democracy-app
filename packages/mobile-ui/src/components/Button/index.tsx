import React from 'react';
import styled from 'styled-components/native';

enum BackgroundColors {
  transparent = 'transparent',
  blue = '#4494d3',
  lightBlue = '#9AC5E7',
  red = '#ec3e31',
}
interface ContainerProps {
  color: keyof typeof BackgroundColors;
  disabled?: boolean;
}

const Container = styled.TouchableOpacity<ContainerProps>`
  background-color: ${({ color, disabled }) =>
    disabled ? '#979797' : BackgroundColors[color]};
  height: 60;
  justify-content: center;
  border-radius: 2;
  margin-top: 11;
`;

enum TextColors {
  blue = '#0076ff',
  white = '#fff',
  red = '#d0021b',
}

interface TextProps {
  color: keyof typeof TextColors;
}
const ButtonText = styled.Text<TextProps>`
  text-align: center;
  color: ${({ color }) => TextColors[color]};
  font-size: 20;
  line-height: 24;
`;

export interface Props {
  text: string;
  onPress: () => void;
  textColor: keyof typeof TextColors;
  backgroundColor?: keyof typeof BackgroundColors;
  disabled?: boolean;
}

export const Button: React.FC<Props> = ({
  text,
  onPress,
  textColor,
  backgroundColor = 'transparent',
  disabled,
}) => (
  <Container
    disabled={disabled}
    testID="Button"
    onPress={onPress}
    color={backgroundColor}>
    <ButtonText testID="ButtonText" color={textColor}>
      {text}
    </ButtonText>
  </Container>
);
