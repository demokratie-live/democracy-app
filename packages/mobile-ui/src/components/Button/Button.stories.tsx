import React from 'react';
import { View } from 'react-native';

import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

import { Button } from '../Button';
import CenterView from '../../decorators/CenterView';

storiesOf('Button', module)
  .addDecorator(CenterView)
  .add('red | transparent', () => (
    <View style={{ width: '100%' }}>
      <Button
        onPress={action('clicked-text')}
        text="Hello Button"
        textColor="red"
      />
    </View>
  ))
  .add('white | blue', () => (
    <View style={{ width: '100%' }}>
      <Button
        onPress={action('clicked-emoji')}
        text="Hello Button"
        textColor="white"
        backgroundColor="blue"
      />
    </View>
  ))
  .add('white | red', () => (
    <View style={{ width: '100%' }}>
      <Button
        onPress={action('clicked-emoji')}
        text="Hello Button"
        textColor="white"
        backgroundColor="red"
      />
    </View>
  ))
  .add('blue | transparent', () => (
    <View style={{ width: '100%' }}>
      <Button
        onPress={action('clicked-emoji')}
        text="Hello Button"
        textColor="blue"
      />
    </View>
  ));
