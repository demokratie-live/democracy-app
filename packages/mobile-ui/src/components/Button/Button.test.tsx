import React from 'react';
import { render } from 'react-native-testing-library';

import { Button, Props } from '../Button';

function renderButton(props: Partial<Props> = {}) {
  const defaultProps: Props = {
    onPress() {
      return;
    },
    text: 'a button text',
    textColor: 'red',
  };
  return render(<Button {...defaultProps} {...props} />);
}

describe('<Button />', () => {
  test('should display a simple button', async () => {
    const { getByTestId } = renderButton();

    const button = await getByTestId('Button');

    expect(button).toMatchSnapshot();
  });
});
