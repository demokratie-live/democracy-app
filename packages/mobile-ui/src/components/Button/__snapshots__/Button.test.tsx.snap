// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should display a simple button 1`] = `
ReactTestInstance {
  "_fiber": FiberNode {
    "_debugHookTypes": null,
    "_debugID": 4,
    "_debugIsCurrentlyTiming": false,
    "_debugNeedsRemount": false,
    "_debugOwner": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 3,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": null,
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": [Circular],
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 1,
      "elementType": [Function],
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": <Text>
          a button text
        </Text>,
        "onPress": [Function],
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": <Text>
          a button text
        </Text>,
        "onPress": [Function],
      },
      "ref": null,
      "return": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 1,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 1,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": null,
          "_debugSource": null,
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": [Circular],
          "child": null,
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 0,
          "elementType": null,
          "expirationTime": 1073741823,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": null,
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": null,
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": FiberRootNode {
            "callbackExpirationTime": 0,
            "callbackNode": null,
            "containerInfo": Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "isHidden": false,
                          "tag": "TEXT",
                          "text": "a button text",
                        },
                      ],
                      "isHidden": false,
                      "props": Object {
                        "children": "a button text",
                      },
                      "rootContainerInstance": [Circular],
                      "tag": "INSTANCE",
                      "type": "Text",
                    },
                  ],
                  "isHidden": false,
                  "props": Object {
                    "children": <Text>
                      a button text
                    </Text>,
                    "testID": "button",
                  },
                  "rootContainerInstance": [Circular],
                  "tag": "INSTANCE",
                  "type": "View",
                },
              ],
              "createNodeMock": [Function],
              "tag": "CONTAINER",
            },
            "context": Object {},
            "current": [Circular],
            "finishedExpirationTime": 0,
            "finishedWork": null,
            "firstBatch": null,
            "firstPendingTime": 0,
            "hydrate": false,
            "interactionThreadID": 1,
            "lastPendingTime": 0,
            "memoizedInteractions": Set {},
            "pendingChildren": null,
            "pendingContext": null,
            "pendingInteractionMap": Map {},
            "pingCache": null,
            "pingTime": 0,
            "tag": 0,
            "timeoutHandle": -1,
          },
          "tag": 3,
          "treeBaseDuration": 0,
          "type": null,
          "updateQueue": Object {
            "baseState": null,
            "firstCapturedEffect": null,
            "firstCapturedUpdate": null,
            "firstEffect": null,
            "firstUpdate": Object {
              "callback": null,
              "expirationTime": 1073741823,
              "next": null,
              "nextEffect": null,
              "payload": Object {
                "element": <Button
                  onPress={[Function]}
                >
                  <Text>
                    a button text
                  </Text>
                </Button>,
              },
              "priority": 97,
              "suspenseConfig": null,
              "tag": 0,
            },
            "lastCapturedEffect": null,
            "lastCapturedUpdate": null,
            "lastEffect": null,
            "lastUpdate": Object {
              "callback": null,
              "expirationTime": 1073741823,
              "next": null,
              "nextEffect": null,
              "payload": Object {
                "element": <Button
                  onPress={[Function]}
                >
                  <Text>
                    a button text
                  </Text>
                </Button>,
              },
              "priority": 97,
              "suspenseConfig": null,
              "tag": 0,
            },
          },
        },
        "child": [Circular],
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 0,
        "firstEffect": [Circular],
        "index": 0,
        "key": null,
        "lastEffect": [Circular],
        "memoizedProps": null,
        "memoizedState": Object {
          "element": <Button
            onPress={[Function]}
          >
            <Text>
              a button text
            </Text>
          </Button>,
        },
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": FiberRootNode {
          "callbackExpirationTime": 0,
          "callbackNode": null,
          "containerInfo": Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "isHidden": false,
                        "tag": "TEXT",
                        "text": "a button text",
                      },
                    ],
                    "isHidden": false,
                    "props": Object {
                      "children": "a button text",
                    },
                    "rootContainerInstance": [Circular],
                    "tag": "INSTANCE",
                    "type": "Text",
                  },
                ],
                "isHidden": false,
                "props": Object {
                  "children": <Text>
                    a button text
                  </Text>,
                  "testID": "button",
                },
                "rootContainerInstance": [Circular],
                "tag": "INSTANCE",
                "type": "View",
              },
            ],
            "createNodeMock": [Function],
            "tag": "CONTAINER",
          },
          "context": Object {},
          "current": [Circular],
          "finishedExpirationTime": 0,
          "finishedWork": null,
          "firstBatch": null,
          "firstPendingTime": 0,
          "hydrate": false,
          "interactionThreadID": 1,
          "lastPendingTime": 0,
          "memoizedInteractions": Set {},
          "pendingChildren": null,
          "pendingContext": null,
          "pendingInteractionMap": Map {},
          "pingCache": null,
          "pingTime": 0,
          "tag": 0,
          "timeoutHandle": -1,
        },
        "tag": 3,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": Object {
          "baseState": Object {
            "element": <Button
              onPress={[Function]}
            >
              <Text>
                a button text
              </Text>
            </Button>,
          },
          "firstCapturedEffect": null,
          "firstCapturedUpdate": null,
          "firstEffect": null,
          "firstUpdate": null,
          "lastCapturedEffect": null,
          "lastCapturedUpdate": null,
          "lastEffect": null,
          "lastUpdate": null,
        },
      },
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": null,
      "tag": 0,
      "treeBaseDuration": 0,
      "type": [Function],
      "updateQueue": null,
    },
    "_debugSource": null,
    "actualDuration": 0,
    "actualStartTime": -1,
    "alternate": null,
    "child": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 5,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": [Circular],
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 6,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": null,
        "child": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 7,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": [Circular],
          "_debugSource": null,
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": null,
          "child": FiberNode {
            "_debugHookTypes": null,
            "_debugID": 8,
            "_debugIsCurrentlyTiming": false,
            "_debugNeedsRemount": false,
            "_debugOwner": null,
            "_debugSource": null,
            "actualDuration": 0,
            "actualStartTime": -1,
            "alternate": null,
            "child": null,
            "childExpirationTime": 0,
            "dependencies": null,
            "effectTag": 0,
            "elementType": null,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": "a button text",
            "memoizedState": null,
            "mode": 0,
            "nextEffect": null,
            "pendingProps": "a button text",
            "ref": null,
            "return": [Circular],
            "selfBaseDuration": 0,
            "sibling": null,
            "stateNode": Object {
              "isHidden": false,
              "tag": "TEXT",
              "text": "a button text",
            },
            "tag": 6,
            "treeBaseDuration": 0,
            "type": null,
            "updateQueue": null,
          },
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 0,
          "elementType": "Text",
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "children": "a button text",
          },
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "children": "a button text",
          },
          "ref": null,
          "return": [Circular],
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": Object {
            "children": Array [
              Object {
                "isHidden": false,
                "tag": "TEXT",
                "text": "a button text",
              },
            ],
            "isHidden": false,
            "props": Object {
              "children": "a button text",
            },
            "rootContainerInstance": Object {
              "children": Array [
                Object {
                  "children": Array [
                    [Circular],
                  ],
                  "isHidden": false,
                  "props": Object {
                    "children": <Text>
                      a button text
                    </Text>,
                    "testID": "button",
                  },
                  "rootContainerInstance": [Circular],
                  "tag": "INSTANCE",
                  "type": "View",
                },
              ],
              "createNodeMock": [Function],
              "tag": "CONTAINER",
            },
            "tag": "INSTANCE",
            "type": "Text",
          },
          "tag": 5,
          "treeBaseDuration": 0,
          "type": "Text",
          "updateQueue": null,
        },
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 1,
        "elementType": [Function],
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "children": "a button text",
        },
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {
          "children": "a button text",
        },
        "ref": null,
        "return": [Circular],
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": Component {
          "_reactInternalFiber": [Circular],
          "_reactInternalInstance": Object {},
          "context": Object {},
          "props": Object {
            "children": "a button text",
          },
          "refs": Object {},
          "state": null,
          "updater": Object {
            "enqueueForceUpdate": [Function],
            "enqueueReplaceState": [Function],
            "enqueueSetState": [Function],
            "isMounted": [Function],
          },
        },
        "tag": 1,
        "treeBaseDuration": 0,
        "type": [Function],
        "updateQueue": null,
      },
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 0,
      "elementType": "View",
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": <Text>
          a button text
        </Text>,
        "testID": "button",
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": <Text>
          a button text
        </Text>,
        "testID": "button",
      },
      "ref": null,
      "return": [Circular],
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "isHidden": false,
                "tag": "TEXT",
                "text": "a button text",
              },
            ],
            "isHidden": false,
            "props": Object {
              "children": "a button text",
            },
            "rootContainerInstance": Object {
              "children": Array [
                [Circular],
              ],
              "createNodeMock": [Function],
              "tag": "CONTAINER",
            },
            "tag": "INSTANCE",
            "type": "Text",
          },
        ],
        "isHidden": false,
        "props": Object {
          "children": <Text>
            a button text
          </Text>,
          "testID": "button",
        },
        "rootContainerInstance": Object {
          "children": Array [
            [Circular],
          ],
          "createNodeMock": [Function],
          "tag": "CONTAINER",
        },
        "tag": "INSTANCE",
        "type": "View",
      },
      "tag": 5,
      "treeBaseDuration": 0,
      "type": "View",
      "updateQueue": null,
    },
    "childExpirationTime": 0,
    "dependencies": null,
    "effectTag": 1,
    "elementType": [Function],
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {
      "children": <Text>
        a button text
      </Text>,
      "testID": "button",
    },
    "memoizedState": null,
    "mode": 0,
    "nextEffect": null,
    "pendingProps": Object {
      "children": <Text>
        a button text
      </Text>,
      "testID": "button",
    },
    "ref": null,
    "return": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 3,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": null,
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": [Circular],
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 1,
      "elementType": [Function],
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": <Text>
          a button text
        </Text>,
        "onPress": [Function],
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": <Text>
          a button text
        </Text>,
        "onPress": [Function],
      },
      "ref": null,
      "return": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 1,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 1,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": null,
          "_debugSource": null,
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": [Circular],
          "child": null,
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 0,
          "elementType": null,
          "expirationTime": 1073741823,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": null,
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": null,
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": FiberRootNode {
            "callbackExpirationTime": 0,
            "callbackNode": null,
            "containerInfo": Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "isHidden": false,
                          "tag": "TEXT",
                          "text": "a button text",
                        },
                      ],
                      "isHidden": false,
                      "props": Object {
                        "children": "a button text",
                      },
                      "rootContainerInstance": [Circular],
                      "tag": "INSTANCE",
                      "type": "Text",
                    },
                  ],
                  "isHidden": false,
                  "props": Object {
                    "children": <Text>
                      a button text
                    </Text>,
                    "testID": "button",
                  },
                  "rootContainerInstance": [Circular],
                  "tag": "INSTANCE",
                  "type": "View",
                },
              ],
              "createNodeMock": [Function],
              "tag": "CONTAINER",
            },
            "context": Object {},
            "current": [Circular],
            "finishedExpirationTime": 0,
            "finishedWork": null,
            "firstBatch": null,
            "firstPendingTime": 0,
            "hydrate": false,
            "interactionThreadID": 1,
            "lastPendingTime": 0,
            "memoizedInteractions": Set {},
            "pendingChildren": null,
            "pendingContext": null,
            "pendingInteractionMap": Map {},
            "pingCache": null,
            "pingTime": 0,
            "tag": 0,
            "timeoutHandle": -1,
          },
          "tag": 3,
          "treeBaseDuration": 0,
          "type": null,
          "updateQueue": Object {
            "baseState": null,
            "firstCapturedEffect": null,
            "firstCapturedUpdate": null,
            "firstEffect": null,
            "firstUpdate": Object {
              "callback": null,
              "expirationTime": 1073741823,
              "next": null,
              "nextEffect": null,
              "payload": Object {
                "element": <Button
                  onPress={[Function]}
                >
                  <Text>
                    a button text
                  </Text>
                </Button>,
              },
              "priority": 97,
              "suspenseConfig": null,
              "tag": 0,
            },
            "lastCapturedEffect": null,
            "lastCapturedUpdate": null,
            "lastEffect": null,
            "lastUpdate": Object {
              "callback": null,
              "expirationTime": 1073741823,
              "next": null,
              "nextEffect": null,
              "payload": Object {
                "element": <Button
                  onPress={[Function]}
                >
                  <Text>
                    a button text
                  </Text>
                </Button>,
              },
              "priority": 97,
              "suspenseConfig": null,
              "tag": 0,
            },
          },
        },
        "child": [Circular],
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 0,
        "firstEffect": [Circular],
        "index": 0,
        "key": null,
        "lastEffect": [Circular],
        "memoizedProps": null,
        "memoizedState": Object {
          "element": <Button
            onPress={[Function]}
          >
            <Text>
              a button text
            </Text>
          </Button>,
        },
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": FiberRootNode {
          "callbackExpirationTime": 0,
          "callbackNode": null,
          "containerInfo": Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "isHidden": false,
                        "tag": "TEXT",
                        "text": "a button text",
                      },
                    ],
                    "isHidden": false,
                    "props": Object {
                      "children": "a button text",
                    },
                    "rootContainerInstance": [Circular],
                    "tag": "INSTANCE",
                    "type": "Text",
                  },
                ],
                "isHidden": false,
                "props": Object {
                  "children": <Text>
                    a button text
                  </Text>,
                  "testID": "button",
                },
                "rootContainerInstance": [Circular],
                "tag": "INSTANCE",
                "type": "View",
              },
            ],
            "createNodeMock": [Function],
            "tag": "CONTAINER",
          },
          "context": Object {},
          "current": [Circular],
          "finishedExpirationTime": 0,
          "finishedWork": null,
          "firstBatch": null,
          "firstPendingTime": 0,
          "hydrate": false,
          "interactionThreadID": 1,
          "lastPendingTime": 0,
          "memoizedInteractions": Set {},
          "pendingChildren": null,
          "pendingContext": null,
          "pendingInteractionMap": Map {},
          "pingCache": null,
          "pingTime": 0,
          "tag": 0,
          "timeoutHandle": -1,
        },
        "tag": 3,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": Object {
          "baseState": Object {
            "element": <Button
              onPress={[Function]}
            >
              <Text>
                a button text
              </Text>
            </Button>,
          },
          "firstCapturedEffect": null,
          "firstCapturedUpdate": null,
          "firstEffect": null,
          "firstUpdate": null,
          "lastCapturedEffect": null,
          "lastCapturedUpdate": null,
          "lastEffect": null,
          "lastUpdate": null,
        },
      },
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": null,
      "tag": 0,
      "treeBaseDuration": 0,
      "type": [Function],
      "updateQueue": null,
    },
    "selfBaseDuration": 0,
    "sibling": null,
    "stateNode": Component {
      "_reactInternalFiber": [Circular],
      "_reactInternalInstance": Object {},
      "context": Object {},
      "props": Object {
        "children": <Text>
          a button text
        </Text>,
        "testID": "button",
      },
      "refs": Object {},
      "state": null,
      "updater": Object {
        "enqueueForceUpdate": [Function],
        "enqueueReplaceState": [Function],
        "enqueueSetState": [Function],
        "isMounted": [Function],
      },
    },
    "tag": 1,
    "treeBaseDuration": 0,
    "type": [Function],
    "updateQueue": null,
  },
}
`;
